### PR TITLE
More fixes for gcc, and also starting vbcc compatibility.

### PR DESCRIPTION
**Pull request by adamel**

### CODE REVIEW

File: Nodes/Brev.c [hunk 1]
```diff
@@ -660,7 +661,7 @@ int updatenextletter(int user) {
 }
 
 void sparabrev(void) {
-  BPTR fh, lock = NULL;
+  BPTR fh, lock = 0;
```
Comment (punktniklas) [inline Nodes/Brev.c]: Does this really have to be changed from NULL to 0. It is a pointer (sort of..)

File: Server/NiKHost2.c [hunk 2]
```diff
@@ -597,7 +598,7 @@ void textinfo(struct RexxMsg *mess) {
                                 ts->tm_mon + 1, ts->tm_mday, ts->tm_hour,
                                 ts->tm_min);
                         break;
-                defualt :
```
Comment (punktniklas) [inline Server/NiKHost2.c]: Interesting that this actually compiled. :-)

File: Tools/SetNodeState/SetNodeState.c [hunk 3]
```diff
@@ -8,11 +8,12 @@
...
 struct Library *NiKomBase;
 
-void main(int argc, char *argv[]) {
+int main(int argc, char *argv[]) {
+  int ret = 0;
   int nodnr, state = 0, i;
   if(argc < 2) {
     puts("Usage: SetNodeState <nodenr> [SERCLOSED] [NOANSWER] [LOGOUT]");
-    return;
+    return 1;
```
Comment (punktniklas) [inline Tools/SetNodeState/SetNodeState.c]: This should probably be 10 (RETURN_ERROR in dos.h)

File: Nodes/Brev.c (hunk 1 above)
Comment (adamel) [inline Nodes/Brev.c]: I have not seen it before, but VBCC seems to be very strict about NULL pointer vs 0, and I guess technically it's correct.

File: Server/NiKHost2.c (hunk 2 above)
Comment (adamel) [inline Server/NiKHost2.c]: Well, you can have labels, gotos, and other loop constructs in really weird ways in switch statements: https://en.wikipedia.org/wiki/Duff%27s_device
vbcc actually complained about the "defualt" label being unused.

File: Tools/SetNodeState/SetNodeState.c (hunk 3 above)
Comment (adamel) [inline Tools/SetNodeState/SetNodeState.c]: Fixed.

**Pull request merged**